plugin/trino-clickhouse/src/main/java/io/trino/plugin/clickhouse/ClickHouseClient.java

```diff
@@ -253,6 +256,11 @@ public ClickHouseClient(
                         .build());
     }
 
+    private String getClusterInfo()
```

Comment:
Adding `ON CLUSTER` to all queries is error-prone. ClickHouse doesn't support configuring the cluster at connection level with JDBC property or something?

Reply:
https://clickhouse.com/docs/sql-reference/distributed-ddl

But yes, it's not that simple, as you can read on https://clickhouse.com/docs/engines/table-engines/special/distributed

In short, you have to create/alter  tables on each node using the `ON CLUSTER ...` , and then create a "virtual" distributed table using the "real" tables as data source.

It may be required to add this distributed table when creating the real ones, I'm not (yet) used to clickhouse cluster mode.